virtwho/register.py

```diff
@@ -1,5 +1,487 @@
-"""
-class SubscriptionManager()
-class RHSMAPI()
-class SatelliteCLI()
-"""
+import re
+import time
+
+from virtwho import logger, FailException
+from virtwho.configure import virtwho_ssh_connect
+from virtwho.configure import guest_ssh_connect
+from virtwho.configure import get_register_handler
+from virtwho.settings import config
+
+
+class SubscriptionManager:
+    """"""
+
+    def __init__(self, mode, register_type):
+        """
+        Define the global variables.
+        :param mode: the hypervisor mode.
+            (esx, xen, hyperv, rhevm, libvirt, kubevirt, local)
+        :param register_type: the subscription server. (rhsm, satellite)
+        """
+        self.mode = mode
+        self.register_type = register_type
+        self.ssh_virtwho = virtwho_ssh_connect(self.mode)
+        self.ssh_guest = guest_ssh_connect(self.mode)
+        self.rhsm_conf = '/etc/rhsm/rhsm.conf'
+        self.rhsm_conf_backup = '/backup/rhsm.conf'
+
+        self.register_section = get_register_handler(register_type)
+        self.server = self.register_section.server
+        self.username = self.register_section.username
+        self.password = self.register_section.password
+        self.port = self.register_section.port
+        self.prefix = self.register_section.prefix
+        self.default_org = self.register_section.default_org
+        if self.register_type == 'satellite':
+            self.secondary_org = self.register_section.secondary_org
+            self.activation_key = self.register_section.activation_key
```

Comment:
how about:
```
    def __init__(self, host, username, password, port, register_type='rhsm', org=None, activation_key=None):
        self.ssh = SSHConnect(host=host, user=username, pwd=password, port=port)
        self.register = get_register_handler(register_type)
        self.server = self.register.server
        self.username = self.register.username
        self.password = self.register.password
        self.port = self.register.port
        self.prefix = self.register.prefix
        self.org = org or self.register.default_org
        self.activation_key = activation_key or self.register.activation_key

```

Reply:
Looks very good and have involved to the file with some changes.